src/api/test-controller/index.js

```diff
@@ -83,7 +82,10 @@ export default class TestController {
     // await t2.click('#btn3');   // <-- without check it will set callsiteWithoutAwait = null, so we will lost tracking
     _createExtendedPromise (promise, callsite) {
         const extendedPromise     = promise.then(identity);
-        const markCallsiteAwaited = () => this.callsitesWithoutAwait.delete(callsite);
+        const markCallsiteAwaited = () => {
+            if (this.testRun.observedCallsites)
```

Comment:
You can use the Elvis operator to simplify this expression:
```ts
this.testRun.observedCallsites?.callsitesWithoutAwait.delete(callsite);
```

Reply:
And maybe it's a good idea to extract `observedCallsites` to a separate variable to simplify the closure.

Reply:
Besides, if you create `observedCallsites` in the TestRun constructor, do we really need to check its existence?